[DOCS] Support for ECS 1.11 indicator data (#1444)
elastic/security-docs: docs/getting-started/advanced-setting.asciidoc

[[update-threat-intel-indices]]
== Update default Elastic Security threat intelligence indices

> The `securitySolution:defaultThreatIndex` advanced setting defines threat intelligence indices that {elastic-sec} features use when querying for ingested threat indicators. The setting controls features that query threat intelligence indices, such as the Threat Intelligence view on the Overview page and indicator match rules. One or more threat intelligence indices can be defined; the `logs-ti*` index pattern is defined by default and used by {agent} integrations.

Reviewer: second sentence: we can add investigation time enrichments here as well.

Reviewer: third sentence: they can add multiple index patterns separated by commas

> The `securitySolution:defaultThreatIndex` advanced setting defines threat intelligence indices that {elastic-sec} features use when querying for ingested threat indicators. The setting controls features that query threat intelligence indices, such as the Threat Intelligence view on the Overview page, indicator match rules, and the alert enrichment query. One or more threat intelligence indices can be defined; multiple indices must be separated by commas. By default, the `logs-ti*` index pattern is defined and used by {agent} integrations.

Reviewer 2: ```suggestion
The `securitySolution:defaultThreatIndex` advanced setting specifies threat intelligence indices that {elastic-sec} features query for ingested threat indicators. This setting affects features that query threat intelligence indices, such as the Threat Intelligence view on the Overview page, indicator match rules, and the alert enrichment query. You can specify one or more threat intelligence indices; multiple indices must be separated by commas. By default, only the `logs-ti*` index pattern is included. 
```
I suggested to remove "and used by {agent} integrations" because Elastic agent integrations aren't mentioned earlier in the paragraph, so I thought it was confusing to add it at the end. As a relatively naive user it left me wondering if agent integrations are in the category of features that query the threat intelligence indices for threat indicators, or if they were another category of thing that also somehow uses the same setting.

Author: I'm completely on board with all of the edits, but do want to keep the bit about Elastic agent integrations. In 7.x, the default threat intel index was populated with Filebeat data (as opposed to data from agent integrations) and named differently (`filebeat-*`) so I think it's necessary to include this distinction.

> The `securitySolution:defaultThreatIndex` advanced setting specifies threat intelligence indices that {elastic-sec} features query for ingested threat indicators. This setting affects features that query threat intelligence indices, such as the Threat Intelligence view on the Overview page, indicator match rules, and the alert enrichment query. You can specify one or more threat intelligence indices; multiple indices must be separated by commas. By default, only the `logs-ti*` index pattern is specified. {agent} integrations use this index pattern.

Reviewer 3: Does the last sentence mean that the `logs-ti*` pattern is required? If so, maybe be a little more explicit that the user shouldn't remove or overwrite it if they want to use Elastic Agent integrations.

Author: Yes, it's required for using Elastic Agent integrations. How about:
By default, only the `logs-ti*` index pattern is specified. Do not remove or overwrite this index pattern, as it's used by {agent} integrations.

Reviewer 3: LGTM!

> IMPORTANT: Threat intelligence indices aren't required to be ECS compatible when used in an indicator match rule. However, we strongly recommend compatibility if you'd like for your alerts to be enriched with relevant threat indicator information. When searching for threat indicator data, indicator match rules uses the threat indicator path specified in the *Indicator prefix override* advanced setting. Visit <<rule-ui-advanced-params, Configure advanced rule settings>> for more information.

Reviewer 2: ```suggestion
IMPORTANT: Threat intelligence indices aren't required to be ECS compatible for use in indicator match rules. However, we strongly recommend compatibility if you want your alerts to be enriched with relevant threat indicator information. When searching for threat indicator data, indicator match rules use the threat indicator path specified in the *Indicator prefix override* advanced setting. Visit <<rule-ui-advanced-params, Configure advanced rule settings>> for more information.
```

Reviewer 3: +1 to @benironside's edit, and also hyphenate "ECS-compatible"

> IMPORTANT: Threat intelligence indices aren't required to be ECS compatible when used in an indicator match rule. However, we recommend compatibility if you'd like for your alerts to be enriched with relevant threat indicator information. When querying for threat indicator data, indicator match rules uses the threat indicator path specified in the *Indicator prefix override* advanced setting. By default, this setting looks for fields prefixed by `threat.indicator`. More information about this setting is available at <<rule-ui-advanced-params, Configure advanced rule settings>>.

Reviewer: Second sentence: I'd dare to say that we **strongly** recommend ECS compatibility (we have some experimental features in the pipeline that may work with ECS threat indices only)
